statsforecast/core.py

```diff
@@ -1546,6 +1551,113 @@ def plot(
             palette="tab20b",
         )
 
+    def save(
+        self,
+        path: Union[os.PathLike, str],
+        file_name: Union[os.PathLike, str, None] = None,
+        max_size: str = "50MB",
+        trim: bool = True,
+        trim_attr: list = ["fcst_fitted_values_", "cv_fitted_values_"],
+    ):
+        """Function that will save StatsForecast class with certain settings to make it
+        reproducible.
+
+        Parameters
+        ----------
+        path: Union[os.PathLike, str]
+            Path of the folder where the pickle should be saved to
+        file_name: Union[os.PathLike, str, None] = None
+            Name of the file, if None then datetime name will be given
+        max_size: str = "50MB"
+            StatsForecast class should not exceed this size, available byte naming:
+            ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB']
+        trim: bool = True
+            Trim any boiler attributes
+        trim_attr:list = None
+            Specify which attributes to trim
+        """
+
+        # Removing unnecessary attributes
+        if trim:
+            for attr in trim_attr:
+                # remove unnecessary attributes here
+                self.__dict__.pop(attr, None)
+
+        # Will be used to find the size of the fitted models
+        # Never expecting anything higher than GB (even that's a lot')
+        bytes_hmap = {
+            "B": 1,
+            "KB": 2**10,
+            "MB": 2**20,
+            "GB": 2**30,
+            "TB": 2**40,
+            "PB": 2**50,
+            "EB": 2**60,
+            "ZB": 2**70,
+            "YB": 2**80,
+        }
+
+        max_size = max_size.replace(" ", "")
+        match = re.match(r"(\d+\.\d+|\d+)(\w+)", max_size)
+        if not match or match[2] not in bytes_hmap.keys():
+            warnings.warn("Couldn't parse your max_size, default 50MB will be set")
+            cap_size = 50.0 * bytes_hmap["MB"]
+        else:
+            m_size = float(match[1])
+            key_ = match[2]
+            cap_size = m_size * bytes_hmap[key_]
+
+        sf_size = sys.getsizeof(self)
```

Comment:
This isn't a reliable way to get the size of the full object (the only way would be using `pickle.dumps` and getting it's size), since it doesn't look into nested attributes. Maybe we can just remove the size restriction and let the user decide if they want to keep it or not after it's been saved

Reply:
fixed with `len(pickle.dumps(sf))`